synapse/lib/types.py

```diff
@@ -630,6 +637,60 @@ def norm(self, valu, oldval=None):
             self._raiseBadValu(valu)
         return valu, {}
 
+class PropValuType(DataType):
+    def __init__(self, tlib, name, **info):
+        DataType.__init__(self, tlib, name, **info)
+        # TODO figure out what to do about tlib vs core issues
+        self._getPropNorm = getattr(tlib, 'getPropNorm', None)
+        self._getPropRepr = getattr(tlib, 'getPropRepr', None)
+
+    def norm(self, valu, oldval=None):
+        # if it's already a str, we'll need to split it into its two parts to norm.
+        if s_compat.isstr(valu):
+            return self._norm_str(valu, oldval=oldval)
+
+        if not islist(valu):
+            self._raiseBadValu(valu, mesg='Expected str or list/tuple.')
+
+        return self._norm_list(valu)
+
+    def _norm_str(self, text, oldval=None):
+
+        text = text.strip()
+
+        if not text:
+            self._raiseBadValu(text, mesg='No text left after strip().')
+
+        if '=' not in text:
+            self._raiseBadValu(text, mesg='Provalu is missing a =')
+
+        valu = text.split('=', 1)
+
+        return self._norm_list(valu)
+
+    def _norm_list(self, valu, oldval=None):
+        if len(valu) != 2:
+            self._raiseBadValu(valu=valu, mesg='PropValu requires two values to norm.')
+
+        prop, valu = valu
+
+        try:
+            nvalu, nsubs = self._getPropNorm(prop, valu, oldval=oldval)
+        except s_common.BadTypeValu as e:
+            logger.exception('Failed to norm PropValu.')
+            self._raiseBadValu(valu, mesg='Unable to norm PropValu', prop=prop)
```

Comment:
Do we need the logger above, since the exception will log almost the same thing?

Reply:
We don't have a super-good way to chain the exceptions the way I'd like too with the helper and python2 support required.  This at least logs what the original norm failure was before wrapping it separately.

Reply:
ok that's fine then